### DISCUSSION TITLE
Event-Based configuration?

### DISCUSSION THREAD

**Discussion by dotzenith** [zed-industries/zed#14230] [Feature Requests]
Hi!

I recently started using Zed and I've been having a fun time! While my experience has been positive, I do find myself missing an event driven config.

As a basic example, I use this autocommand in neovim:
```lua
autocmd({ "BufEnter", "FocusGained", "InsertLeave", "CmdlineLeave", "WinEnter" }, {
   pattern = "*",
   callback = function()
      if vim.o.nu and vim.api.nvim_get_mode().mode ~= "i" then
         vim.opt.relativenumber = true
      end
   end,
})

autocmd({ "BufLeave", "FocusLost", "InsertEnter", "CmdlineEnter", "WinLeave" }, {
   pattern = "*",
   callback = function()
      if vim.o.nu then
         vim.opt.relativenumber = false
         vim.cmd "redraw"
      end
   end,
})
```

This allows me to seamlessly switch between relative line numbers in normal mode, and absolute line numbers in insert mode. As far as I know, there's no way to accomplish something like this in Zed as of now.

I know that the config file allows configuration based on filetype, and keybinds have scopes, but I don't know if that is enough.

So the question here is, are there any plans to support this sort of configuration? The example I gave is very basic and not a deal-breaker, but I think there would be a lot of cool ways to extend Zed if there was a way to write an event driven config. Maybe this could be accomplished via the extensions api? though I don't think you can do that right now either.

**Comment by m-ocean-it** [comment]
related https://github.com/zed-industries/zed/issues/8325

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
